src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -675,7 +675,11 @@ protected bool TryGetValue(FieldsToFetch.FieldToFetch fieldToFetch, Document doc
                 if (_loadedDocuments.TryGetValue(docId, out var doc) == false)
                 {
                     using (_loadScope = _loadScope?.Start() ?? _projectionScope?.For(nameof(QueryTimingsScope.Names.Load)))
+                    {
                         _loadedDocuments[docId] = doc = LoadDocument(docId);
+                        if (doc != null)
+                            doc.IgnoreDispose = true;
```

Comment:
I have a feeling that this is too broad. Where is the code that is disposing them? Are we disposing somewhere else the `_loadedDocuments` ? Can it cause some memory pressure now because of lack of dispose?

Reply:
That is the responsability of the context, that will be disposed when they are done.

Reply:
Not always, sometimes we are releasing sooner. Where it was previously disposed?

Reply:
This was disposed in the streaming, after we run output the document, we'll dicpose it.

Reply:
More precisely it was disposed here:

https://github.com/ravendb/ravendb/blob/d8d91a5ebe53350ac26970ce615fe3ba51337206/src/Raven.Server/Documents/Queries/StreamDocumentQueryResult.cs#L16-L17